lib/puck/jar.rb

```diff
@@ -140,6 +148,14 @@ def create
                 zipfileset dir: spec[:base_path], prefix: File.join(JAR_GEM_HOME, spec[:versioned_name])
               end
             end
+
+            merge_archives.each do |archive, target_path|
+              if target_path
+                zipfileset src: archive, prefix: target_path, excludes: SIGNATURE_FILES
+              else
+                zipfileset src: archive, excludes: SIGNATURE_FILES
+              end
+            end
```

Comment:
You could probably simplify the code a bit by treating the JRuby jars as any other merge archives (but apply them last).